onnxruntime/core/graph/graph_utils.h

```diff
@@ -57,6 +57,11 @@ const ONNX_NAMESPACE::TensorProto* GetConstantInitializer(const Graph& graph, co
 void ReplaceInitializer(Graph& graph, const std::string& original_name, const ONNX_NAMESPACE::TensorProto& initializer,
                         bool check_outer_scope = true);
 
+/** Returns a pointer to the corresponding TensorProto for a name if it is an initializer. 
+@param check_outer_scope If true and the graph is a subgraph, check parent graph/s for 'name' if not found in 'graph'.
+*/
+const onnx::TensorProto* ReturnInitializedValue(const Graph& graph, const std::string& name, bool check_outer_scope);
+
```

Comment:
Maybe just put this locally in the memcpy transformer for now as I'm not aware of any other use case for it.

Reply:
Well is there value in introducing this to graph_utils even if there is no immediate use-case ? Moving it locally to MemCpy seems rather unnecessary especially since it does align with the overall theme of graph_utils....

Reply:
It aligns with the theme but if it's only used by one thing it adds clutter. Up to you. If you leave it here maybe call it GetInitializer given it is the same as GetConstantInitializer sans the constant check. 

---
In reply to: [303695110](https://github.com/microsoft/onnxruntime/pull/1413#discussion_r303695110) [](ancestors = 303695110)

Reply:
I d prefer to keep this in graph_utils for now. I made the change to the name of the method as suggested. Thanks!

Reply:
Can we combine the two methods to one? With a boolean that specifies whether we should constant/overridable check.
Seems confusing to have two methods, especially given it is quite nuanced for most people what it means to have a non-constant initializer.

Reply:
Will scope this to just tranformer_memcpy.cc for now if it is confusing.